### ISSUE TITLE
How to use VPD for classification?

### ISSUE THREAD

**Issue by BingliangLi** [wl-zhao/VPD#11]
Hi, thanks for open source the code for VPD!
I'm very interested in your work, and I wonder if I can use VPD for zero-shot image classification, here is my plan:

1. Get embeddings(`class_embeddings.pth`) for different class labels.
2. Get the cross-attention map for a given image with `c_crossattn`(which did not go through a text adapter).
3. Find the most responsive embedding on the cross-attention map, which is the classification result.

My question is: how to get the logits for each embedding, so I can use softmax to get the prediction score? A code example would be great.

I managed to obtain `outs` (`x_c1, x_c2, x_c3, x_c4`) and `class_embeddings`. What should I do next? Any help would be appreciated.

**Comment by wl-zhao** [comment]
Hi, thanks for your interest in VPD.

Your idea to perform zero-shot image classification is interesting. However, I think VPD is not suitable for zero-shot inference because it is not a diffusion-based framework anymore. Fine-tuning is required to adapt the pre-trained knowledge to downstream tasks in VPD. 

However, there are some recent works that leverage the text-to-image diffusion model for zero-shot image classification, such as https://arxiv.org/abs/2303.15233 and https://arxiv.org/abs/2303.16203. I think maybe these papers are helpful to you. Please feel free to further discussions.

**Comment by BingliangLi** [comment]
Thank you for your explanation!

I found the papers you mentioned as well. Indeed, they are more related to my problem, but they haven't been open-sourced yet. I will try some experiments with VPD (finetune or zero-shot) and update this thread when I have the results. I'll close the issue for now.

Thanks again, VPD is a great work!
